cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,391 @@
+                    kafkaAssembly.getMetadata().getName(),
+                    Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
+            result.isDeployed = true;
+
+            if (kafkaAssembly.getSpec().getKafka().getJmxOptions().getAuthentication() instanceof KafkaJmxAuthenticationPassword) {
+                result.isJmxAuthenticated = true;
+            }
+
+            result.setResources(spec.getResources());
+
+            String image = spec.getImage();
+            if (image == null) {
+                image = System.getenv().getOrDefault(STRIMZI_DEFAULT_JMXTRANS_IMAGE, "strimzi/jmxtrans:latest");
+            }
+            result.setImage(image);
+
+            result.setOwnerReference(kafkaAssembly);
+        }
+
+        return result;
+    }
+
+    public Deployment generateDeployment(ImagePullPolicy imagePullPolicy, List<LocalObjectReference> imagePullSecrets) {
+        if (!isDeployed()) {
+            return null;
+        }
+
+        DeploymentStrategy updateStrategy = new DeploymentStrategyBuilder()
+                .withType("RollingUpdate")
+                .withRollingUpdate(new RollingUpdateDeploymentBuilder()
+                        .withMaxSurge(new IntOrString(1))
+                        .withMaxUnavailable(new IntOrString(0))
+                        .build())
+                .build();
+
+        return createDeployment(
+                updateStrategy,
+                Collections.emptyMap(),
+                Collections.emptyMap(),
+                getMergedAffinity(),
+                getInitContainers(imagePullPolicy),
+                getContainers(imagePullPolicy),
+                getVolumes(),
+                imagePullSecrets
+        );
+    }
+
+    /**
+     * Generates the string'd config that the JmxTrans deployment needs to run. It is configured by the user in the yaml
+     * and this method will convert that into the config the JmxTrans understands.
+     * @param spec The JmxTrans that was defined by the user
+     * @param numOfBrokers number of kafka brokers
+     * @return the jmx trans config file that targets each broker
+     */
+    private String generateJMXConfig(JmxTransSpec spec, int numOfBrokers) {
+        Servers servers = new Servers();
+        servers.servers = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        String headlessService = KafkaCluster.headlessServiceName(cluster) + "." + namespace + ".svc";
+        for (int brokerNumber = 0; brokerNumber < numOfBrokers; brokerNumber++) {
+            String brokerServiceName = KafkaCluster.externalServiceName(clusterName, brokerNumber) + "." + headlessService;
+            servers.servers.add(convertSpecToServers(spec, brokerServiceName));
+        }
+        try {
+            return mapper.writeValueAsString(servers);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
Either propagate the exception (wrap it in a `RuntimeException` if you're sure there's no way for it to be thrown in practic), or at least log it properly.